maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java

```diff
@@ -55,9 +56,7 @@
 
     private static final String DEFAULT_LANGUAGE = "java";
 
-    private List<Parameter> parameters;
-
-    private Map<String, Parameter> parameterMap;
+    private final ArrayList<Parameter> parameters;
```

Comment:
I tend to favour interfaces when possible:
```
private final List<Parameter> parameters;
```

Reply:
Well, this is a matter of taste IMHO and we had chat about this with @michael-o as well.
I see value of interfaces only at "public surfaces" (like non-private method parameters and return values). For method local variables (and private members) I tend to always do it like this:
```
public method int doSomethig() {
  HashMap<T1, T2> tempMap = new HashMap<>();
  ...
  return tempMap.size();
}
```
doing it as `Map<T1, T2> tempMap = new HashMap<>();` would be just 2 imports vs 1 import w/o any real benefit.

In this case, `parameters` is internal matter of this class. Nowhere is directly exposed (just a copy of it), so IMHO it is completely it's own matter. What benefit would be gained here by using interface?

Reply:
It leads to think that the chosen implementation is actually important, which isn't really the case and any `List` implementation should work.  Looser coupling of some kind.
No big deal though, I certainly won't veto for such minor and arguable things.

Reply:
Agreed :smile: But re "leads to think" IMHO quite the opposite: ArrayList is the "first java list impl that comes to mind", so to say, I would raise eye-browse if I'd see there some non-common List implementation. By clearly stating ArrayList, I think am telling "literally any list would do", same thing you say.